apis/python/src/tiledbsoma/_point_cloud.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) 2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+"""
+Implementation of a SOMA Point Cloud
+"""
+
+import warnings
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import pyarrow as pa
+import somacore
+from somacore import CoordinateSpace, CoordinateTransform, options
+from typing_extensions import Self
+
+from ._constants import SOMA_JOINID, SPATIAL_DISCLAIMER
+from ._dataframe import Domain
+from ._read_iters import TableReadIter
+from ._types import OpenTimestamp
+from .options import SOMATileDBContext
+
+_UNBATCHED = options.BatchSize()
+
+
+class PointCloud(somacore.PointCloud):
+    """A specialized SOMA DataFrame for storing collections of points in
+    multi-dimensional space.
+
+    The ``PointCloud`` class is designed to efficiently store and query point data,
+    where each point is represented by coordinates in one or more spatial dimensions
+    (e.g., x, y, z) and may have additional columns for associated attributes.
+
+    Lifecycle:
+        Experimental.
+    """
+
+    __slots__ = ()
+
+    @classmethod
+    def create(
+        cls,
+        uri: str,
+        *,
+        schema: pa.Schema,
+        index_column_names: Sequence[str] = (SOMA_JOINID, "x", "y"),
+        axis_names: Sequence[str] = ("x", "y"),
+        domain: Optional[Domain] = None,
+        platform_config: Optional[options.PlatformConfig] = None,
+        context: Optional[SOMATileDBContext] = None,
+        tiledb_timestamp: Optional[OpenTimestamp] = None,
+    ) -> Self:
+        """Creates a new ``PointCloud`` at the given URI.
+
+        The schema of the created point cloud will include a column named
```

Comment:
```suggestion
        The schema of the created point-cloud dataframe will include a column named
```

Reply:
```suggestion
        The schema of the created point cloud dataframe will include a column named
```
I like the dataframe, but the hyphen on point cloud feels wrong.